server/graphql/v1/mutations/orders.js

```diff
@@ -231,6 +231,45 @@ const hasPaymentMethod = order => {
   }
 };
 
+export const deletePaymentMethod = async (collective, fingerprint, transaction) => {
+  // Check if the credit card is already saved
+  const oldPaymentMethod = await models.PaymentMethod.findOne(
+    {
+      where: {
+        CollectiveId: collective.id,
+        service: 'stripe',
+        saved: true,
+        data: {
+          fingerprint: fingerprint,
+        },
+      },
+    },
+    { transaction },
+  );
+
+  if (oldPaymentMethod) {
+    await oldPaymentMethod.destroy();
+  }
+
+  return oldPaymentMethod;
+};
+
+export const updateExistingOrdersToNewPaymentMethod = async (paymentMethod, oldPaymentMethod, transaction) => {
+  if (paymentMethod && oldPaymentMethod) {
+    // Update all existing orders with the new payment method
+    await models.Order.update(
+      { PaymentMethodId: paymentMethod.id },
+      {
+        where: {
+          PaymentMethodId: oldPaymentMethod.id,
+          status: { [Op.notIn]: ['PAID', 'CANCELLED', 'EXPIRED', 'DISPUTED', 'REFUNDED'] },
+        },
+      },
+      { transaction },
```

Comment:
As per https://sequelize.org/api/v6/class/src/model.js~model#instance-method-update, the transaction is expected in 2nd object. Typescript would have detected this.

Reply:
Corrected. 👍🏽